tests/components/homeassistant/triggers/test_time.py

```diff
@@ -877,3 +877,228 @@ async def test_if_at_template_limited_template(
     await hass.async_block_till_done()
 
     assert "is not supported in limited templates" in caplog.text
+
+
+async def test_if_fires_using_weekday_single(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+    service_calls: list[ServiceCall],
+) -> None:
+    """Test for firing on a specific weekday."""
+    # Freeze time to Monday, January 2, 2023 at 4:59:00
+    now = dt_util.now()
+    monday_before_trigger = now.replace(
+        year=2023, month=1, day=2, hour=4, minute=59, second=0, microsecond=0
+    )
+    monday_trigger = now.replace(
+        year=2023, month=1, day=2, hour=5, minute=0, second=0, microsecond=0
+    )
+
+    freezer.move_to(monday_before_trigger)
```

Comment:
```suggestion
    monday_trigger = now.replace(
        year=2023, month=1, day=2, hour=5, minute=0, second=0, microsecond=0
    )
    freezer.move_to(monday_trigger)
```
I am not sure I understand why we have `monday_before_trigger` and `monday_trigger`, but then do `monday_trigger + timedelta(seconds=1)`. Couldn't we just have `monday_trigger`?

Reply:
By the way, we could use the `@pytest.mark.freeze_time("2023-01-02 05:00:00")` decorator, and then just use `now` + timedeltas instead of `now().replace()` like `tuesday_trigger = now + timedelta(days=1)`.

I think that would make the tests easier to read, but I am fine with either approache.